Client/Coordinators/Coordinator.swift

```diff
@@ -0,0 +1,36 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+protocol Coordinator {
+    var id: UUID { get }
+    var childCoordinators: [Coordinator] { get }
+    var router: Router { get }
+
+    func addChild(_ coordinator: Coordinator)
```

Comment:
nit: I got used from Swift standard library to name methods with the verb and arguments inside paranthesis. In this case, I would do something like `add(child coordinator: Coordinator)`, `remove(child coordinator: Coordinator)`.